### PR TITLE
PHPCS: JSON GET Endpoints Part 2

### LINKED ISSUE #21956: PHPCS: JSON Endpoints List
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-comments-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-embeds-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-media-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-media-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-media-v1-2-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-type-taxonomies-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-types-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-roles-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-shortcodes-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-terms-endpoint.php",
	"projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-users-endpoint.php",
```

### PR DESCRIPTION
**Pull request by sdixon194**
Fixes #21956

#### Changes proposed in this Pull Request:

* phpcs changes for a bunch of JSON endpoint files

#### Jetpack product discussion

n/a
#### Does this pull request change what data or activity we track or use?

no
#### Testing instructions:

* Proof it?
*

### CODE REVIEW

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-comments-endpoint.php [hunk 1]
```diff
@@ -149,57 +211,58 @@ function callback( $path = '', $blog_id = 0, $object_id = 0 ) {
...
+					return new WP_Error( 'unauthorized', 'User cannot read non-approved comments', 403 );
+				}
+				if ( 'unapproved' === $args['status'] ) {
+					$status       = 'hold';
+					$count_status = 'moderated';
+				} elseif ( 'all' === $args['status'] ) {
+					$status       = 'all';
+					$count_status = 'total_comments';
+				} else {
+					$status = $count_status = $args['status']; // phpcs:ignore Squiz.PHP.DisallowMultipleAssignments.Found
```
Comment (jeherve) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-comments-endpoint.php:256]: Why not breaking it down here?

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-type-taxonomies-endpoint.php [hunk 2]
```diff
@@ -41,26 +64,26 @@ function callback( $path = '', $blog_id = 0, $post_type = 'post' ) {
 
 		$this->localize_initial_taxonomies( $post_type );
 
-		$args = $this->query_args();
+		$args = $this->query_args(); // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
Comment (jeherve) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-type-taxonomies-endpoint.php:67]: If this isn't used, can we get rid of it? If not, maybe you could add a comment as to why?

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-types-endpoint.php [hunk 3]
```diff
@@ -49,7 +70,7 @@ function callback( $path = '', $blog_id = 0 ) {
 			$this->load_theme_functions();
 		}
 
-		$args = $this->query_args();
+		$args = $this->query_args(); // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
Comment (jeherve) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-types-endpoint.php:73]: Same as above

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php [hunk 4]
```diff
@@ -159,42 +193,46 @@ function callback( $path = '', $blog_id = 0 ) {
...
+			}
 
 			$query['meta_query'] = array( $meta );
 		}
 
 		if (
 			isset( $args['sticky'] )
 		&&
-			( $sticky = get_option( 'sticky_posts' ) )
+			( $sticky = get_option( 'sticky_posts' ) ) // phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition.Found, Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure
```
Comment (jeherve) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php:224]: Do you think you could move the assignment before the condition?

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-comments-endpoint.php (hunk 1 above)
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-comments-endpoint.php]: Fear of change, mostly: 78869eefea1a077b78fe102bfd854937179df0c0

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php (hunk 4 above)
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php]: Let me know if dc5fb9e055537b69460d5318b4c5e7b919da778c works

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-type-taxonomies-endpoint.php (hunk 2 above)
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-type-taxonomies-endpoint.php]: dc5fb9e055537b69460d5318b4c5e7b919da778c

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-types-endpoint.php (hunk 3 above)
Comment (sdixon194) [reply projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-post-types-endpoint.php]: dc5fb9e055537b69460d5318b4c5e7b919da778c

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-embeds-endpoint.php [hunk 5]
```diff
@@ -1,60 +1,77 @@
...
+			require_once ABSPATH . WPINC . '/class-oembed.php';
 		}
 
 		global $wp_embed;
 		$oembed = _wp_oembed_get_object();
 
-		foreach( $wp_embed->handlers as $priority => $handlers ) {
-			foreach( $handlers as $handler ) {
-				if ( ! empty( $handler['regex'] ) )
+		foreach ( $wp_embed->handlers as $priority => $handlers ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-embeds-endpoint.php:61]: If the key is unused, why not just remove it?
```suggestion
		foreach ( $wp_embed->handlers as $handlers ) {
```

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-media-v1-2-endpoint.php [hunk 6]
```diff
@@ -63,11 +77,11 @@ function callback( $path = '', $blog_id = 0 ) {
 			return $response;
 		}
 
-		foreach ( $media_list as $index => $media_item ) {
-			// expose `revision_history` object for each image
+		foreach ( $media_list as $index => $media_item ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-media-v1-2-endpoint.php:80]:
```suggestion
		foreach ( $media_list as $media_item ) {
```

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php [hunk 7]
```diff
@@ -159,42 +193,47 @@ function callback( $path = '', $blog_id = 0 ) {
...
 
 			$query['meta_query'] = array( $meta );
 		}
 
+		$sticky = get_option( 'sticky_posts' );
 		if (
 			isset( $args['sticky'] )
 		&&
-			( $sticky = get_option( 'sticky_posts' ) )
+			( $sticky )
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-endpoint.php:225]:
```suggestion
			$sticky
```

File: projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php [hunk 8]
```diff
@@ -80,32 +94,32 @@ function callback( $path = '', $blog_id = 0 ) {
...
 			}
 
-			// Normalize post_type
-			if ( 'any' == $args['type'] ) {
+			// Normalize post_type.
+			if ( 'any' === $args['type'] ) {
 				$whitelisted_post_types = $site->get_whitelisted_post_types();
 
-				if ( isset( $args['exclude_private_types'] ) && $args['exclude_private_types'] == true ) {
+				if ( isset( $args['exclude_private_types'] ) && true == $args['exclude_private_types'] ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
```
Comment (anomiex) [inline projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php:110]:
```suggestion
				if ( ! empty( $args['exclude_private_types'] ) ) {
```

**Comment by sdixon194** [comment]
r243098-wpcom

**Pull request merged**